zebra-consensus/src/checkpoint.rs

```diff
@@ -56,10 +49,155 @@ struct QueuedBlock {
 /// has an old chain fork. (Or sends us a bad block.)
 type QueuedBlockList = Vec<QueuedBlock>;
 
-/// A block height verification range.
+/// A `CheckpointVerifier`'s current progress verifying the chain.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum Progress<HeightOrHash> {
+    /// We have not verified any blocks yet.
+    // rustc is smart enough to know that this variant isn't used for hashes,
+    // because we always check the height before getting the hash.
+    #[allow(dead_code)]
```

Comment:
I'm not sure if this is the reason -- it looks like the variant is constructed, in `CheckpointVerifier::new`, but that code is cfg-gated so it isn't counted.

Reply:
`CheckpointVerifier::new` is cfg-gated because it's only used in the tests, so dev builds generate an unused code lint. A similar reason applies here.

Reply:
Hmm, actually that's not a particularly clean fix. Using `allow(dead_code)` on `CheckpointVerifier::new` means we only have to allow the lint in one place, and it applies to everything used by that function as well.

Reply:
Fixed in d40b2f46e6f10f81cf18c41416ba3fb991f37323